### PR TITLE
[Mobile Payments] Add card images to Refund Confirmation screen

### LINKED ISSUE #6179: [Mobile Payments] Show brand icons for cards on Refund screen
We should look into displaying the card brand icons instead of names on the Refund Confirmation screen. 

Looking at [Mastercard brand guidelines](https://brand.mastercard.com/brandcenter/branding-requirements/mastercard.html) again, they mention "Mastercard" being acceptable next to the last 4 digits only for devices that support text-only:
Screen Shot 2022-02-14 at 11 11 33

_Originally posted by @koke in https://github.com/woocommerce/woocommerce-ios/pull/6130#issuecomment-1038898789_

### PR DESCRIPTION
**Pull request by joshheald**
Closes: #6179 

Will not merge until after 8.6 freezes – there's too much to tweak with the designs still.

### Description

Adds card brand images to the Refund Via row in the Refund Confirmation screen. This is to fulfil brand guidelines from card networks, and to make it easier to identify the card in question when performing a refund.

Card images were taken from the `Hardware` target and are not final – e.g. we do not have an `Interac` image, and the `Visa` image is outdated (it should not have a gold flash any more.)

Apologies for the length - it's mostly SVGs

### Testing instructions

1. In the orders tab, open an order which was paid for with a Card Present payment
2. Tap `Issue Refund`
3. Select some of the order to refund, and tap `Next`
4. Observe that the card image is shown alongside the card details
5. With VoiceOver on, hear that the payment method cell is read out as `Visa card ending four two four two`

### Screenshots

Card image on refund confirmation

card-brand-screenshot-interac | card-brand-screenshot-visa | card-brand-screenshot-amex
card-brand-screenshot-diners | card-brand-screenshot-discover | card-brand-screenshot-jcb
card-brand-screenshot-mastercard | card-brand-screenshot-unionpay | card-brand-screenshot-unknown

#### Visa 
This should not have the gold flash seen in the screenshot above any more. [The Visa brand guidelines page](https://merchantsignageap.visa.com/apac_brand_guidelines) provides SVGZ files, but I can't find a free way to turn those in to SVGs for use in the app – Adobe CC will do it I think. Please could you get the up to date logo as an SVG for me?

#### Interac
I've used the [Interac logo SVG from WikiMedia](https://commons.wikimedia.org/wiki/File:InteracLogo.svg) – I'm not sure whether we actually want to do that, but it was provided by Interac and it's in line with their [brand guidelines](https://interacposcentre.ca/images/INTERAC_Brand_Guidelines_Vendors_Apr2021_FINAL.pdf), except that it includes a ® stamp.

#### Discover and Diners club
The brand guidelines suggest that the images we have (taken from WCPay) are outdated. There are EPS files for both [available to download on discoverglobalsinage.com](https://discoverglobalsignage.com/free-signage-logos). I think the correct ones are the Acceptance Marks using RGB colour. Similar to Visa, I don't have any tools suitable for doing the conversion to SVG myself: would you be able to convert them please?

#### Amex & UnionPay
I wasn't able to find useful branding guidelines for these two card networks, but I think what we have here is appropriate. An internal Amex document suggested that the `AMEX` square should only be used at very small sizes, and I think that the size we have here is appropriate.

#### Brand guidelines used
Where I could find them, I checked our use of these marks against the brand guidelines that are available.

- [Discover and Diners Club](https://discoverglobalsignage.com/uploads/09-12-21_03:59_DGN_BG_10052021_AccMk.pdf)
- [Mastercard](https://brand.mastercard.com/brandcenter/branding-requirements/mastercard.html)
- [Visa](https://spponeimages.azureedge.net/prod/a8c3df05-c491-404b-b41b-e6bfe0d77e3b2a_VDB-experience-guidelines-2018.pdf)
- [JCB](https://www.global.jcb/en/about-us/brand-concept/agreement/index.html)

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: Hardware/Hardware/CardReader/CardBrand.swift [hunk 1]
```diff
@@ -18,6 +18,8 @@ public enum CardBrand: String, CaseIterable, Codable {
     /// Diners Club card
     case dinersClub
 
+    //TODO: Add Interac here
```
Comment (joshheald) [inline Hardware/Hardware/CardReader/CardBrand.swift:21]: Will do this in a separate PR

**Comment by joe-keenan** [comment]
Hi @joshheald!

What you have looks pretty good, just thought I’d tweak the size so the cards are a little bit smaller, so they line up with the height of the two lines of text more neatly, and so that there’s an even amount of space around all sides of them.

I’ve also rebuilt the Discover and Visa images and used an online converter to do the Diner’s Club one (I’ve only just requested Adobe – haven’t had need of it until now!). You should be able to export SVGs directly from Figma.

I’ve mocked up a few examples, with redlines, and put the new images here: https://www.figma.com/file/Y3QRKOQJjKDdcmjxrSslpRyB/Orders---Refunds-%26-Cancelations?node-id=3854%3A6867

Let me know if you need anything else!

**Comment by koke** [comment]
> [The Visa brand guidelines page](https://merchantsignageap.visa.com/apac_brand_guidelines) provides SVGZ files, but I can't find a free way to turn those in to SVGs for use in the app

For future reference, it seems SVGZ is just gzipped SVG. You can rename the files to `.svg.gz`, double click, and Finder will extract them to SVG.

File: WooCommerce/Classes/ViewRelated/Orders/Order Details/Issue Refunds/RefundConfirmationViewModel.swift [hunk 2]
```diff
@@ -322,4 +354,35 @@ private extension WCPayCardBrand {
...
+    func cardBrandName() -> String {
+        switch self {
+        case .amex:
+            return "American Express"
+        case .diners:
+            return "Diners Club"
+        case .discover:
+            return "Discover"
+        case .interac:
+            return "interac"
```
Comment (koke) [inline WooCommerce/Classes/ViewRelated/Orders/Order Details/Issue Refunds/RefundConfirmationViewModel.swift:367]:
```suggestion
            return "Interac"
```

**Comment by joe-keenan** [comment]
Thanks @koke! That’s really handy to know.

File: WooCommerce/Classes/ViewRelated/CardPresentPayments/WCPayCardBrand+icons.swift [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+extension WCPayCardBrand {
+    /// Icon that represents the card brand in SVG format
+    ///
+    var icon: UIImage? {
+        UIImage(named: iconName)
+    }
+
+    /// Icon name to be used for the card icon UIImage
+    var iconName: String {
+        switch self {
```
Comment (koke) [inline WooCommerce/Classes/ViewRelated/CardPresentPayments/WCPayCardBrand+icons.swift:13]: This is fine, but I wonder if it's worth simplifying here and returning `"card-brand-\(rawValue)"` instead

File: WooCommerce/Classes/ViewRelated/Orders/Order Details/Issue Refunds/RefundConfirmationViewModel.swift [hunk 4]
```diff
@@ -302,6 +314,24 @@ private extension WCPayCardBrand {
...
+
+        guard let brandNameRange = localizedDescription.range(of: cardBrandName()),
+              let last4Range = localizedDescription.range(of: last4)
+        else {
+            return attributedLocalizedDescription
+        }
+
+        let brandNameNSRange = NSRange(brandNameRange, in: localizedDescription)
+        let last4NSRange = NSRange(last4Range, in: localizedDescription)
+        attributedLocalizedDescription.setAttributes([.accessibilitySpeechLanguage: "en-US"], range: brandNameNSRange)
```
Comment (koke) [inline WooCommerce/Classes/ViewRelated/Orders/Order Details/Issue Refunds/RefundConfirmationViewModel.swift:329]: I though that was a nice touch until I actually tested it. It sounds off when it uses a different voice and pronunciation. While the card brands are mostly English words, they're often adapted (mispronounced?) to the local language, so it would sound off with American English. Forcing `en-US` might even sound odd in British/Canadian/Australian/... settings.

**Review by koke — APPROVED** [review]
Left some minor comments but the implementation is looking pretty good to me

**Comment by joshheald** [comment]
> Hi @joshheald!
> 
> What you have looks pretty good, just thought I’d tweak the size so the cards are a little bit smaller, so they line up with the height of the two lines of text more neatly, and so that there’s an even amount of space around all sides of them.
> 
> I’ve also rebuilt the Discover and Visa images and used an online converter to do the Diner’s Club one (I’ve only just requested Adobe – haven’t had need of it until now!). You should be able to export SVGs directly from Figma.
> 
> I’ve mocked up a few examples, with redlines, and put the new images here: https://www.figma.com/file/Y3QRKOQJjKDdcmjxrSslpRyB/Orders---Refunds-%26-Cancelations?node-id=3854%3A6867
> 
> Let me know if you need anything else!

Here's the screenshots with the three updated cards

Visa | Discover | Diners
---|---|---
Visa | Discover | Diners


Diners looks a little distorted at these small sizes unfortunately.

**Comment by joshheald** [comment]
https://user-images.githubusercontent.com/2472348/155178624-a669ab50-220f-4175-ab98-cbaa5235170f.mp4

@koke I got the voiceover localization to work

**Comment by joe-keenan** [comment]
Yeah, that’s not great. I liked their earlier one with just the logo much better.
The others are looking acceptable, though. I think the padding is good now.

File: WooCommerce/Classes/ViewRelated/Orders/Order Details/Issue Refunds/RefundConfirmationViewModel.swift (hunk 4 above)
Comment (joshheald) [reply WooCommerce/Classes/ViewRelated/Orders/Order Details/Issue Refunds/RefundConfirmationViewModel.swift:329]: https://user-images.githubusercontent.com/2472348/155321169-20504fbe-9da5-489f-9698-14d2a43a11b8.mp4

Agreed, it sounds better without. I'll remove this bit.

File: WooCommerce/Classes/ViewRelated/CardPresentPayments/WCPayCardBrand+icons.swift (hunk 3 above)
Comment (joshheald) [reply WooCommerce/Classes/ViewRelated/CardPresentPayments/WCPayCardBrand+icons.swift:13]: I used the switch so that we'd provide a placeholder image even if we add a new card brand without providing an image for it in future... so I think I'll leave this one if that's OK 😊

File: WooCommerce/Classes/ViewRelated/Orders/Order Details/Issue Refunds/RefundConfirmationViewModel.swift (hunk 2 above)
Comment (joshheald) [reply WooCommerce/Classes/ViewRelated/Orders/Order Details/Issue Refunds/RefundConfirmationViewModel.swift:367]: Good spot
